custom_components/acwd/acwd_api.py

```
@@ -40,15 +50,32 @@ def _get_hidden_fields(self, soup):
-        response = self.session.get(self.base_url)
+        # Use retry logic for network requests
+        try:
+            response = self._retry_request(
+                self.session.get,
+                self.base_url,
+                max_retries=3,
+                retry_delay=2.0,
+                timeout=30,
+                headers={'User-Agent': USER_AGENT}
+            )
+        except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)